### DISCUSSION TITLE
No valid CABL license found

### DISCUSSION THREAD

**Discussion by bladepoint6969** [vscode-abl/vscode-abl#151] [Q&A] [answered]
I'm trying to get CABL scanning working in a small test project, but whenever I make a change to a file, I get an `Unable to instantiate rule eu.rssw.antlr.proparse.checks.FindFirst - No valid CABL license found` error, despite having copied in what I believe is the correct license into the extension settings.

In my workspaces `.vscode/settings.json` I have the `abl.cablLicense` setting set to the license key I pulled from our SonarQube deployment (marked as `sonar.oe.license` in the Administration menu. Not sure what I'm doing wrong here.

**Comment by gquerret** [comment, marked as answer]
You have to go to the [license management website](https://cabl.riverside-software.fr) ; from there, you can download the VSCode license key (which is a much shorter key).
